### ISSUE TITLE
Lots of zero-size files in ENOSP situation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#63]
```
This is just a FYI.   We've seen this odd behavior, when running out of disk 
space.  In this particular LevelDB instance, we had ~500.000 files, of which 
~386.000 were zero-length.

Let me know if you want more info for further investigation if you find this 
relevant.

Kresten


^count  ^byte size
8986  2100454 
8986  2099630 
8988  2097214 
   3  2031616 
8986  1141855 
8986   817817 
8987   796603 
8987   786279 
8988   679909 
8986   675593 
8986   622029 
8987   549499 
8986   545361 
8988   543867 
8986   483052 
   2   458752 
8987   270322 
   2   196608 
8987   173884 
   5    65536 
386412        0 


```

---

Original issue reported on code.google.com by `kresten....@gmail.com` on 22 Dec 2011 at 4:39

**Comment by GoogleCodeExporter** [comment]
```
Hi Kresten,

Thanks for the report.

(1) Can you send me the names of some of the zero length files?  That will
help track down which part of the code was generating them.

(2) if there is a non-empty LOG file in the database directory,
the contents of that file could also help.

(3) Do you know what you were doing when this happened?
E.g., repeatedly opening the DB, or opening the DB once and
doing lots of writes to it, or opening the DB and just reading from it.
```

---

Original comment by `san...@google.com` on 18 Jan 2012 at 6:11

**Comment by GoogleCodeExporter** [comment]
```
Fixed in change 3c8be108bfb5
```

---

Original comment by `san...@google.com` on 25 Jan 2012 at 11:15
- Changed state: **Fixed**
